homeassistant/components/plugwise/coordinator.py

```diff
@@ -0,0 +1,36 @@
+"""DataUpdateCoordinator for Plugwise."""
+from datetime import timedelta
+
+import async_timeout
+from plugwise import Smile
+from plugwise.exceptions import XMLDataMissingError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DEFAULT_TIMEOUT, DOMAIN, LOGGER
+
+
+class PlugwiseDataUpdateCoordinator(DataUpdateCoordinator[bool]):
+    """Class to manage fetching Plugwise data from single endpoint."""
+
+    def __init__(self, hass: HomeAssistant, api: Smile) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=api.smile_name or DOMAIN,
```

Comment:
"or DOMAIN"  is not needed, there's always a `smile_name` available.

There can be more than one Smile in a Core system, that's why we've used `api.smile_name` to separate the DUC's. But I'm not sure that's needed? If not, `name` can be DOMAIN.

Reply:
> "or DOMAIN" is not needed, there's always a smile_name available.

In that case, it is incorrectly typed in the upstream library, as it is initialized as `None`.

Reply:
Ok understood, in the backend we could init `smile_name` as "None", would that work?

Reply:
@bouwew It is already `None`, which is the problem. It means `smile_name` becomes `None | Unknown` right now.